.github/workflows/validate-lease.yaml

```diff
@@ -0,0 +1,38 @@
+name: Validate Lease
+
+on:
+  pull_request:
+    types:
+      # default
+      - opened
+      - synchronize
+      - reopened
+      # re-run if base branch is changed, since previous merge commit may generate incorrect diff
+      - edited
+
+permissions:
+  contents: read
+
+jobs:
+  validate-folder-structure:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+
+      - name: Install dependencies
+        run: |
+          cd .github
+          npm install
+
+      - name: Validate lease files
```

Comment:
For consistency with our other workflows (and some useful helpers), you'll want to use the `github-script` action to run your code.  Example:

https://github.com/Azure/azure-rest-api-specs/blob/743867f3c674453fd2c9f1e930ac4ff2385ee4f7/.github/workflows/arm-incremental-typespec.yaml#L51-L57